src/embedded/embeddedClickEvents.ts

```diff
@@ -3,38 +3,58 @@ import {
   EmbeddedMessageElementsButton,
   IterableEmbeddedMessage
 } from '../embedded';
+import { ErrorHandler, trackEmbeddedClick } from '..';
 
 const embeddedManager = new EmbeddedManager();
-export const handleElementClick = (message: IterableEmbeddedMessage) => {
+export const handleElementClick = (
+  message: IterableEmbeddedMessage,
+  errorCallback?: ErrorHandler
+) => {
   const clickedUrl =
     message?.elements?.defaultAction?.data?.trim() ||
     message?.elements?.defaultAction?.type ||
     null;
   embeddedManager.handleEmbeddedClick(message, null, clickedUrl);
-  embeddedManager.trackEmbeddedClick(message, '', clickedUrl ? clickedUrl : '');
+  trackEmbeddedClick({
+    messageId: message.metadata.messageId,
+    buttonIdentifier: '',
+    clickedUrl: clickedUrl ? clickedUrl : '',
+    errorCallback
+  });
 };
 
 export const handleButtonClick = (
   button: EmbeddedMessageElementsButton,
-  message: IterableEmbeddedMessage
+  message: IterableEmbeddedMessage,
+  errorCallback?: ErrorHandler
 ) => {
   const embeddedManager = new EmbeddedManager();
   const clickedUrl = button?.action?.data?.trim() || button?.action?.type || '';
   embeddedManager.handleEmbeddedClick(message, button?.id || null, clickedUrl);
-  embeddedManager.trackEmbeddedClick(message, button?.id || '', clickedUrl);
+  trackEmbeddedClick({
```

Comment:
@mprew97 "I didn't fully understand your last comment. Do you mean that we should call the `errorcallback` function after handling the promise of `trackEmbeddedClick`, and then remove that callback as a parameter inside the `trackEmbeddedClick` method? If we do that, how will the `errorcallback` function work when developers call the `trackEmbeddedClick` method directly from the SDK? If you remember we have this method available as a standalone method which developers could use."

Reply:
Great questions. 

Taking a step back, we only need a callback for the OOTB views. Any time a customer _directly_ calls `trackEmbeddedClick`, they can just do ```trackEmbeddedClick(...).then().catch```. Meaning, they can do whatever they want and they don't need to specify `errorCallback` because they will have access to the result of the promise anyway. 

Since the OOTB views call that method and they don't have access to handle what goes on in the `catch`, we have to allow the developers to determine what to do with the `catch` in the component. So we really only need the callback passed to each OOTB view where the callback just accepts the generic error as it's only param. 

Does that make sense?

Reply:
ok, so we remove the handler from trackembeddedclick and only keep the ones in OOTB views, right?

Reply:
Should we still keep this errorCallback as a constructor param or should we create a method that says, cardView.setErrorHandler(this) ?

Reply:
I'd say constructor param.

and yep, that exactly.

Reply:
Did the suggested changes? I see some circular dependency errors in GitHub actions which I will try to handle tomorrow.

Reply:
Hi @mprew97 Resolved conflicts and did some changes and also fixed circular dependency.